sound/soc/sof/sof-audio.c

```diff
@@ -311,7 +311,7 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 
 	/* return if the widget is in use or if it is already unprepared */
 	if (!swidget->prepared || swidget->use_count > 1)
-		return;
+		goto sink_unprepare;
```

Comment:
commit message, should we add a 

````
Link: https://github.com/thesofproject/linux/issues/4021
````

Reply:
just thinking a bit aloud @bardliao. Im not sure why I used swidget->use_count > 1, IF a widget is in use, its use_count will be > 0 right?

Reply:
makes sense to me, you only want to unprepare if there is exactly one user of the widget. If there is more than one you skip it.

Reply:
right, so we can even get rid of the prepared flag and just use use_count to decide whether we need to prepare or not.

Reply:
are you sure about the change @bardliao ? This will never unprepare.

Reply:
@ranj063 @plbossart It sounds reasonable. Let me do more test.

Reply:
> are you sure about the change @bardliao ? This will never unprepare.

@plbossart I rechecked the code, and I think it should be correct. It will not unprepare and visit the next widget in the path when `swidget->use_count > 0`. So, it will unprepare when swidget->use_count = 0. This is correct, right?